conda-store-server/conda_store_server/app.py

```diff
@@ -200,7 +201,7 @@ class CondaStore(LoggingConfigurable):
     )
 
     database_url = Unicode(
-        "sqlite:///conda-store.sqlite",
+        "sqlite:///" + str(PosixPath.home()) + "/.conda-store/conda-store.sqlite",
```

Comment:
Need to remove `Posix`, it doesn't work on Windows:

```diff
-        "sqlite:///" + str(PosixPath.home()) + "/.conda-store/conda-store.sqlite",
+        "sqlite:///" + str(Path.home()) + "/.conda-store/conda-store.sqlite",
```

Tested on Windows: after rebasing on https://github.com/conda-incubator/conda-store/pull/559 and fixing the Posix thing, the server started (`python -m conda_store_server.server`) and stored state files in the home dir.
<img width="461" alt="Screen Shot 2023-10-01 at 19 42 39" src="https://github.com/conda-incubator/conda-store/assets/10469885/8bdbd850-7301-4c83-8783-096218ddc825">

Reply:
Tested on Linux: works without changes.

```
% ls -1
conda-store.sqlite
conda-store-state
% pwd
/home/nkaretnikov/.conda-store
```

Reply:
Verified that this env builds on Linux and is available after restart (with the same state dir):

```
channels:
- conda-forge
dependencies:
- python
- pip:
  - nothing
- ipykernel
- pytest
- requests
description: ''
name: test-env
prefix: null
variables: null
```

Server was started via `python -m conda_store_server.server --standalone`.

Reply:
Are you sure about that? PositPath is used for URLs because URLs always use forward slashes, even on Windows.

Reply:
See for instance https://github.com/conda-incubator/conda-store/pull/559/commits/80d9ea863aa0acbc4f3420449ee97ad1e1f65254

Reply:
> Are you sure about that? PosixPath is used for URLs because URLs always use forward slashes, even on Windows.

Have you tested on Windows? When I ran it on Windows, it raised an error.

Reply:
Apparently on Windows you have to use pathlib.PurePosixPath instead of pathlib.PosixPath